### ISSUE TITLE
Recorded video jumping when playing it back from flash player

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mondain/red5#15]
```
Hi,
In our application we allow the publishing and playing of live streams from a 
flex client. We also allow that flex client to replay the recorded streams too. 
We found no problem with the live streams, but when we replayed them later on, 
the video was jumping (only on flash player, not on other players such as VLC), 
as if there were gaps on the recorded file.
One of the flex devs here found out that if the stream contained both audio and 
video, as soon as the audio went completely silent, that will produce a gap on 
the video as it was not saving a timestamp. That is a problem for us since we 
rely on the silence level setting for the microphone, and we may send empty 
audio packets frequently. So i modified red5 with the attached patch and its 
been working perfectly for a couple of months already.
Basically, it will make the FLVWriter.writeTag method to perform the writing 
even if the body of the given tag is empty, therefore saving a timestamp on the 
file.
I try to keep the latest code from red5 up to date, but since i've read you are 
planning on releasing 1.0 shortly, maybe its a good idea to merge it into 
trunk. What do you think?
Thanks

What steps will reproduce the problem?
1.
2.
3.

What is the expected output? What do you see instead?


Please use labels and text to provide additional information.


```

Original issue reported on code.google.com by `daccatt...@gmail.com` on 28 Jan 2011 at 6:11

Attachments:
- [patch-gaps-SilenceLevel.txt](https://storage.googleapis.com/google-code-attachments/red5/issue-15/comment-0/patch-gaps-SilenceLevel.txt)
